packages/router/router.ts

```diff
@@ -1606,17 +1606,18 @@ export function createRouter(init: RouterInit): Router {
       "Expected a location on the redirect navigation"
     );
 
-    if (
-      redirect.external &&
-      typeof window !== "undefined" &&
-      typeof window.location !== "undefined"
-    ) {
-      if (replace) {
-        window.location.replace(redirect.location);
-      } else {
-        window.location.assign(redirect.location);
+    // Check if this an external redirect that goes to a new origin
+    if (typeof window?.location !== "undefined") {
+      let currentOrigin = window.location.origin;
+      let newOrigin = new URL(redirect.location, currentOrigin).origin;
+      if (currentOrigin !== newOrigin) {
+        if (replace) {
+          window.location.replace(redirect.location);
+        } else {
+          window.location.assign(redirect.location);
+        }
+        return;
       }
-      return;
```

Comment:
Got rid of the `redirect.external` flag in favor of just doing the external detection here in client-side only code.